File: String/Java/palindrome.java
public class palindrome {
    public static void main(String[] args) {
        String str = "NOON";
        int i = 0, j = str.length() - 1;
        boolean isPalindrome = true;

        while (i < j) {
            if (str.charAt(i) != str.charAt(j)) {
                isPalindrome = false;
                break;
            }
            i++;
            j--;
        }

        if (isPalindrome) {
            System.out.println("The given string is a palindrome");
        } else {
            System.out.println("The given string is not a palindrome");
        }
    }
}

File: String/C++/palindrome.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    string str = "NOON";
    int i = 0, j = str.length() - 1;
    bool isPalindrome = true;

    while (i < j) {
        if (str[i] != str[j]) {
            isPalindrome = false;
            break;
        }
        i++;
        j--;
    }

    if (isPalindrome) {
        cout << "The given string is a palindrome" << endl;
    } else {
        cout << "The given string is not a palindrome" << endl;
    }

    return 0;
}